bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/shared/ScriptedRuleProvider.java

```diff
@@ -60,13 +60,14 @@ public void addRule(Rule rule) {
     }
 
     public void removeRule(String ruleUID) {
-        Rule rule = rules.get(ruleUID);
+        Rule rule = rules.remove(ruleUID);
```

Comment:
Isn't this now duplicating the removal from `rules`?
I'd think it would be cleaner to keep the "get" here and have it only removed in `removeRule(Rule rule)`, wdyt?

Reply:
I thought about that, too. This way we avoid possible race conditions where the method with the id called twice and then calls the removal of the rule a second time. But that’s probably a very rare case, because it would require a bug at another place that calls remove. I‘ll change it as suggested.